lib/Site.js

```diff
@@ -74,6 +77,38 @@ function setExtension(filename, ext) {
   );
 }
 
+/**
+* Creates a function that delays invoking `func` until after `wait` milliseconds have elapsed
+* and the running `func` has resolved/rejected.
+* @param func the promise-returning function to delay,
+*        func should be able to take in an indefinite amount of arguments
```

Comment:
Continuing from https://github.com/MarkBind/markbind-cli/pull/31#discussion_r173388278:

> aaron:
> `func`'s parameter must be an array, right?

> rachx:
> Nope but we can change it that way
> * Currently, it is using `func.apply(context, pendingArgs);` and `pendingArgs` is supposed to be the argument list. So all the delayed functions take in a variable number of arguments `...filePaths`
> * Will you prefer it to be `func.apply(context, [pendingArgs]);` instead?

1. Can we require that the original  `func` take in an array, and limit `arg` of the wrapped function to a single parameter? This doesn't seem like a good use case for *spread syntax* and *rest parameters*. For flexibility, we can add a check for whether `arg` is already an array or a single value.
2. Let's rename `rebuildSourceFiles` to `rebuildAffectedSourceFiles` to better reflect what it does (similar to the existing `regenerateAffectedPages`).

Reply:
To clarify,

>

Reply:
> Does this mean removing the rest parameters in `rebuildSourceFiles` etc and ensuring an array is always passed in, e.g. by using `[pendingArgs]` and updating the jsdocs accordingly?

Yes.

> Does this mean changing `function (...args)` [[code](https://github.com/MarkBind/markbind-cli/pull/31/files#diff-e5f3ecb7effaa21db6153e4647ea22bcR93)] in `delay` to  `function (arg)`. 
>
> So `arg` in `rebuildSourceFiles` (the non-prefixed method) can be a single value or array

Yes.

> Which lines are you referring to?

*This* as in our use case of passing in a single `filePath` for "original" function accepting an array.
1. is not useful because we really won't be calling it as `buildAsset(filePath1, filePath2)` even if we have multiple files. It makes more sense to pass in an array.
2. seems reasonable at first, but flattening defeats the purpose of rest parameters.
  e.g. If we have an "original" function that accepts (`param1`, `param2`), `pendingArgs` becomes `[param1_first, param2_first, param1_second, param2_second]`. So we're already expecting the "original" function to collect it into an array via rest parameters.
3. is not useful for the same reason as 1.